src/dist/replication/lib/duplication/replica_duplicator_manager.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2017-present, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#include <dsn/dist/replication/duplication_common.h>
+#include <dsn/dist/fmt_logging.h>
+
+#include "replica_duplicator_manager.h"
+
+namespace dsn {
+namespace replication {
+
+std::vector<duplication_confirm_entry>
+replica_duplicator_manager::get_duplication_confirms_to_update() const
+{
+    zauto_lock l(_lock);
+
+    std::vector<duplication_confirm_entry> updates;
+    for (const auto &kv : _duplications) {
+        replica_duplicator *duplicator = kv.second.get();
+        duplication_progress p = duplicator->progress();
+        if (p.last_decree != p.confirmed_decree) {
+            dcheck_gt_replica(p.last_decree, p.confirmed_decree);
+            duplication_confirm_entry entry;
+            entry.dupid = duplicator->id();
+            entry.confirmed_decree = p.last_decree;
+            updates.emplace_back(entry);
+        }
+    }
+    return updates;
+}
+
+void replica_duplicator_manager::sync_duplication(const duplication_entry &ent)
```

Comment:
Replica server send sync request through duplication_sync_timer, so all fresh duplication progress are in meta server, replica should accept the progress from meta server. If my understanding is correct, I would have a question that when replica server report those progress?

Reply:
See my previous PR https://github.com/XiaoMi/rdsn/pull/281, the duplication_sync_timer sends duplication_sync_request, which contains the full progress on this replica-server.